src/openforms/js/components/admin/form_design/registrations/objectsapi/ObjectsApiOptionsFormFields.js

```diff
@@ -0,0 +1,360 @@
+  const {loading: objecttypesLoading} = useAsync(async () => {
+    const response = await get(REGISTRATION_OBJECTTYPES_ENDPOINT);
+    if (!response.ok) {
+      throw new Error('Error when loading objecttypes');
+    }
+
+    setAvailableObjecttypes(response.data);
+  });
+
+  const {loading: objecttypeVersionLoading} = useAsync(async () => {
+    if (!objecttype) return;
+
+    const response = await get(
+      REGISTRATION_OBJECTTYPE_VERSIONS_ENDPOINT.replace(
+        '{objecttypeUuid}',
+        availableObjecttypes.find(o => o.url === objecttype).uuid
+      )
+    );
+    if (!response.ok) {
+      throw new Error('Error when loading objecttype versions');
+    }
+
+    setAvailableObjecttypeVersions(response.data);
+  }, [objecttype]);
+
+  return (
+    <Wrapper>
+      <CustomFieldTemplate
+        id="root_objecttype"
+        label={intl.formatMessage({
+          defaultMessage: 'Objecttype',
+          description: 'Objects API registration options "Objecttype" label',
+        })}
+        rawDescription={intl.formatMessage({
+          defaultMessage:
+            'URL that points to the ProductAanvraag objecttype in the Objecttypes API. The objecttype should have the following three attributes: "submission_id", "type" (the type of productaanvraag) and "data" (the submitted form data)',
+          description: 'Objects API registration options "Objecttype" description',
+        })}
+        rawErrors={getFieldErrors(name, index, validationErrors, 'objecttype')}
+        errors={buildErrorsComponent('objecttype')}
+        displayLabel
+      >
+        {objecttypesLoading ? (
+          <Loader />
+        ) : unknownObjecttype ? (
+          <TextInput
+            id="root_objecttype"
+            name="objecttype"
+            value={objecttype}
+            onChange={onFieldChange}
+            allowBlank={true}
+          />
+        ) : (
+          <Select
+            id="root_objecttype"
+            name="objecttype"
+            choices={availableObjecttypes.map(ot => [ot.url, `${ot.name} (${ot.uuid})`])}
+            value={objecttype}
+            onChange={onFieldChange}
+            allowBlank={true}
+          />
+        )}
+      </CustomFieldTemplate>
+      <CustomFieldTemplate
+        id="root_objecttypeVersion"
+        label={intl.formatMessage({
+          defaultMessage: 'Objecttype version',
+          description: 'Objects API registration options "Objecttype version" label',
+        })}
+        rawDescription={intl.formatMessage({
+          defaultMessage: 'Version of the objecttype in the Objecttypes API',
+          description: 'Objects API registration options "Objecttype version" description',
+        })}
+        rawErrors={getFieldErrors(name, index, validationErrors, 'objecttypeVersion')}
+        errors={buildErrorsComponent('objecttypeVersion')}
+        displayLabel
+      >
+        {objecttypeVersionLoading ? (
+          <Loader />
+        ) : unknownObjecttype ? (
```

Comment:
For some reason, when I change objecttype the version field disappears for a moment but I don't see the loading spinner :thinking:

Reply:
It's too fast to be able to start animating (the loader animation has a small delay before appearing).
Maybe the select components should always display, and if it is loading then it's just a matter of having the choices updated by the API call

Reply:
select with `options={[['...', 'loading...']]}` could be a nice compromise?